### ISSUE TITLE
anchorId already registered! error

### ISSUE THREAD

**Issue by NightGrodno** [hakimio/ngx-ui-tour#34]
Hi, 
it's possible to override or disable error in register function?

if (this.anchors[anchorId]) {
    throw new Error('anchorId ' + anchorId + ' already registered!');
}

**Comment by hakimio** [comment]
No, anchor ids have to be unique. However, if you leave anchor id empty (`<div tourAnchor=""></div>`), it will be skipped.

**Comment by NightGrodno** [comment]
I has situation in my tutorial, on a list of dynamically generated field which also can have a same button (in consequential with the same anchor id). I wuld like to show a first instantion of this button on tutorial, and ignore the other one.  
I dosnt hawe a problem whith a lot of error in my console, but it slows loading of my page.
So i wuild like to disable this warning

**Comment by ste2425** [comment]
I know this is a super old ticket, sorry to necro.

However i figured id share my solution as I found this issue yesterday trying to research my issue.

I was facing the same error whilst working with items within nested lists (its a complex UI).

Well my solution was to extend TourService and override its provider. I hooked into the register and unregister methods so i was in control of what tour directives were registered at any point in time. So i can them work around the issue.

Just thought id share the technique, once you have your extended `TourService` you could have any behaviour you desire. Another feature i added was namespaced yours. So instead of just a single flat list of yours, i have groups and when i call start i now provide a key to say which group of yours i want to begin.

For the issue specifically logged here you could do something like this which will just un-register an element before registering it, meaning if you try to register dupes the last one registers is what will be used.

```ts
@Injectable({
  providedIn: 'root'
})
class ExtendedTourService extends TourService {
  // Note anchor should be of type `TourAnchorDirective` but that is not piblicly exported
  // so cannot be set.
  override register(anchorId: string, anchor: any): void {
    console.log('OVERRIDE');
    super.unregister(anchorId);
    return super.register(anchorId, anchor);
  }
}

export const appConfig: ApplicationConfig = {
  providers: [
    {
      provide: TourService,
      useClass: ExtendedTourService
    },
    provideMarkdown(), 
    provideHttpClient(), 
    provideRouter(routes), 
    provideAnimationsAsync()
  ]
};
```

**Comment by ste2425** [comment]
> Or you could just use [duplicateAnchorHandling](https://hakimio.github.io/ngx-ui-tour/md-menu/API#:~:text=duplicateAnchorHandling) setting.

Ahh but there’s the fun in *just* doing that when I can re-invent the wheel. 

In seriousness, I missed that part of the documentation. 

At least the starting a tour based on a namespace doesn’t seem to be re-invention!
